src/pages/SignIn.js

```diff
@@ -0,0 +1,55 @@
+import {useContext, useState} from 'react';
+import {useNavigate} from 'react-router-dom';
+import UserCtx from '../context/user-context';
+
+const SignIn = props => {
+
+    const userCtx = useContext(UserCtx);
+    const navigate = useNavigate();
+    const [userData, setUserData] = useState({
+        userName: '',
+        password: ''
+    });
+    
+    const signOnHandler = event => {
+        event.preventDefault(); 
+        if(!userData.userName || userData.userName.length === 0 || !userData.password || userData.password.lengt === 0){
+            alert('User name and password are mandatory to fill!');
```

Comment:
Forget about alerts.
For now you should have only fields and no action per button click. Ideally - disable button till conditions are met.
You will have a separate task for that.